src/modules/powerrename/dll/PowerRenameExt.cpp

```diff
@@ -195,7 +196,8 @@ DWORD WINAPI CPowerRenameMenu::s_PowerRenameUIThreadProc(_In_ void* pData)
 
 HRESULT __stdcall CPowerRenameMenu::GetTitle(IShellItemArray* /*psiItemArray*/, LPWSTR* ppszName)
 {
-    return SHStrDup(L"PowerRename", ppszName);
+    app_name = GET_RES_STRING_WCHAR(IDS_POWERRENAME);

```

Comment:
We should initialize `app_name` only once.

Reply:
As I changed the implementation to use a wstring instead of a wchar_t* and am no longer allocating any memory using new or malloc, I don't think there would be an issue. In my understanding I cannot see another way of obtaining the string from the resource table instead of using hardcoded strings. Please let me know if you have another approach in mind.

Reply:
Moved it to the constructor.